wordpress-mobile/WordPress-Android
# Update Global Styles endpoint to support non-experimental paths

This updates the Global styles endpoint to use the non-experimental paths for fetching the block editor settings. 

📓  Note that the Jetpack side is still dependent on some backend changes so this PR is still a draft.

**`WordPress-FluxC-Android` PR**: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2049

## To test:
For each of these cases, it'll be enough to open the editor and validate that the correct color options are available. It would also be useful to test using a network monitoring tool to make sure the correct paths are called. This will help ensure you aren't seeing cached data if you're reusing the same site for testing.

- [ ] Self-Hosted Site with WP 5.8 and base Gutenberg version 
    - Expect the request to be made with `__experimental/wp-block-editor/v1`
- [ ] Self-Hosted Site with WP 5.8 and Gutenberg version >= 11.1
    - Expect the request to be made with `/wp-block-editor/v1`
- [ ] WPCom or Jetpack connected site with WP 5.8 and base Gutenberg version 
    - Expect the request to be made with `__experimental/wp-block-editor/v1`
- [ ] WPCom or Jetpack connected site with WP 5.8 and Gutenberg version >= 11.1
    - Expect the request to be made with `/wp-block-editor/v1`

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Check [`WordPress-FluxC-Android` PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2049)

PR submission checklist:
